server/etcdserver/v3_server.go

```diff
@@ -560,6 +564,10 @@ func (s *EtcdServer) RoleGrantPermission(ctx context.Context, r *pb.AuthRoleGran
 }
 
 func (s *EtcdServer) RoleGet(ctx context.Context, r *pb.AuthRoleGetRequest) (*pb.AuthRoleGetResponse, error) {
+	if rootRoleName == r.Role {
```

Comment:
It is better to add auth logic in the auth module.
https://github.com/etcd-io/etcd/blob/main/server/auth/store.go

Reply:
done.